### PR TITLE
Maven publication: adding Maven central repository

### PR DESCRIPTION
**Pull request by DariusIMP**
- Editing the `build.gradle.kts` file in order to include a publication task for Maven central
- Added package signature (which is required for remote publications)
- Edited the CI workflows for both android and jvm publications.

### CODE REVIEW

File: .github/workflows/publish-android.yml [hunk 1]
```diff
@@ -70,7 +70,17 @@ jobs:
...
           fi
 
-      - name: Gradle Publish Android Package
-        run: ./gradlew publishAndroidReleasePublicationToGithubPackagesRepository -Pandroid=true ${{ env.PUB_MODE }}
+      - name: Gradle Publish Android Package to GitHub packages repository
+        run: ./gradlew publishAndroidReleasePublicationToGithubPackagesRepository -PremotePublication=true -Pandroid=true ${{ env.PUB_MODE }}
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          PGP_SIGNING_KEY: ${{ secrets.PGP_SIGNING_KEY }}
+          PGP_SIGNING_PASSWORD: ${{ secrets.PGP_SIGNING_PASSWORD }}
```
Comment (diogomatsubara) [inline .github/workflows/publish-android.yml:78]: Looks like we can use the `secrets.SSH_PRIVATE_KEY` and `secrets.SSH_PASSPHRASE` here instead. Could you try that? I filed a request to have the GPG keys added but they told me to use the same one already in the org secrets.

File: .github/workflows/publish-android.yml [hunk 2]
```diff
@@ -70,7 +70,17 @@ jobs:
...
+
+      - name: Gradle Publish Android Package to Maven Central repository
+        run: ./gradlew publishAndroidReleasePublicationToMavenCentralRepository -PremotePublication=true -Pandroid=true ${{ env.PUB_MODE }}
+        env:
+          ORG_OSSRH_USERNAME: ${{ secrets.ORG_OSSRH_USERNAME }}
+          ORG_OSSRH_PASSWORD: ${{ secrets.ORG_OSSRH_PASSWORD }}
+          PGP_SIGNING_KEY: ${{ secrets.PGP_SIGNING_KEY }}
+          PGP_SIGNING_PASSWORD: ${{ secrets.PGP_SIGNING_PASSWORD }}
```
Comment (diogomatsubara) [inline .github/workflows/publish-android.yml:86]: These will need an update as well

File: .github/workflows/publish-jvm.yml [hunk 3]
```diff
@@ -175,7 +175,17 @@ jobs:
...
+          PGP_SIGNING_KEY: ${{ secrets.PGP_SIGNING_KEY }}
+          PGP_SIGNING_PASSWORD: ${{ secrets.PGP_SIGNING_PASSWORD }}
+
+      - name: Gradle Publish JVM Package to Maven Central repository
+        run: ./gradlew publishJvmPublicationToMavenCentralRepository -PremotePublication=true ${{ env.PUB_MODE }}
+        env:
+          ORG_OSSRH_USERNAME: ${{ secrets.ORG_OSSRH_USERNAME }}
+          ORG_OSSRH_PASSWORD: ${{ secrets.ORG_OSSRH_PASSWORD }}
+          PGP_SIGNING_KEY: ${{ secrets.PGP_SIGNING_KEY }}
+          PGP_SIGNING_PASSWORD: ${{ secrets.PGP_SIGNING_PASSWORD }}
```
Comment (diogomatsubara) [inline .github/workflows/publish-jvm.yml:191]: And here

**Review by diogomatsubara — CHANGES_REQUESTED** [review]

**Review by diogomatsubara — APPROVED** [review]

**Comment by diogomatsubara** [comment]
Should we add the `-PdryRun=true` option to the gradle calls and add an additional input to run it in dry run mode? 
I see the snapshot input already uses the input.live_run so maybe we can change the logic there?

**Comment by diogomatsubara** [comment]
Looks like we'll need the keyid as well when using `gradle`. Once the key id is available in the org secrets, I'll let you know and we can update.

**Comment by diogomatsubara** [comment]
@DariusIMP eclipse has updated the secrets. They also changed the variable names to conform to their internal policy. You can see the variables here: https://github.com/eclipse-zenoh/.eclipsefdn/pull/18. Could you update this PR accordingly?

**Review by diogomatsubara — APPROVED** [review]
LGTM

File: zenoh-kotlin/build.gradle.kts [hunk 4]
```diff
@@ -106,9 +157,32 @@ kotlin {
                     password = System.getenv("GITHUB_TOKEN")
                 }
             }
+            maven {
+                name = "MavenCentral"
+                url = uri(if (project.hasProperty("SNAPSHOT"))
+                    "https://oss.sonatype.org/content/repositories/snapshots/"
+                else
+                    "https://oss.sonatype.org/service/local/staging/deploy/maven2/"
+                )
```
Comment (diogomatsubara) [inline zenoh-kotlin/build.gradle.kts:166]: Shouldn't this be reversed? If it's a snapshot, publish to the staging repository?

Comment (DariusIMP) [reply zenoh-kotlin/build.gradle.kts:166]: I think it's ok as it is, based on the documentation from https://oss.sonatype.org/content/repositories/snapshots/

**Pull request merged**